filebeat/input/filestream/internal/input-logfile/manager.go

```diff
@@ -167,18 +167,21 @@ func (cim *InputManager) Create(config *conf.C) (v2.Input, error) {
 		return nil, err
 	}
 
+	defer func() {
+		//If there is any error creating the input, remove it from the IDs list
+		if retErr != nil {
+			cim.idsMux.Lock()
+			delete(cim.ids, settings.ID)
+			cim.idsMux.Unlock()
+		}
+	}()
+
 	if settings.ID == "" {
 		cim.Logger.Warn("filestream input without ID is discouraged, please add an ID and restart Filebeat")
 	}
 
 	cim.idsMux.Lock()
 	if _, exists := cim.ids[settings.ID]; exists {
-		duplicatedInput := map[string]any{}
-		unpackErr := config.Unpack(&duplicatedInput)
-		if unpackErr != nil {
-			duplicatedInput["error"] = fmt.Errorf("failed to unpack duplicated input config: %w", unpackErr).Error()
-		}
```

Comment:
@AndersonQ I was looking at the code and realised this is dead code... `duplicatedInput` is never used after L180. What was its intended purpose?

Reply:
just looking at that piece of code, I don't remember. Most likely it'd be used to report the error after.